k9mail/src/main/java/com/fsck/k9/controller/NonQresyncExtensionHandler.java

```diff
@@ -47,15 +47,15 @@ int continueSync(MessageDownloader messageDownloader, FlagSyncHelper flagSyncHel
         Map<String, ImapMessage> remoteUidMap = new HashMap<>();
 
         int remoteMessageCount = imapFolder.getMessageCount();
-        int remoteStart = ImapSyncInteractor.getRemoteStart(localFolder, imapFolder);
+        int remoteStart = syncHelper.getRemoteStart(localFolder, imapFolder);
 
         Timber.v("SYNC: About to fetch UIDs for messages %d through %d in folder %s",
                 remoteStart, remoteMessageCount, folderName);
 
         findRemoteMessagesToDownload(localUidMap, remoteMessages, remoteUidMap, remoteStart);
 
         if (account.syncRemoteDeletions()) {
-            imapSyncInteractor.syncRemoteDeletions(findDeletedMessageUids(localUidMap, remoteUidMap));
+            imapSyncInteractor.syncRemoteDeletions(findDeletedMessageUids(localUidMap, remoteUidMap), syncHelper);
```

Comment:
Moved both the delete local messages method and update more messages method into `SyncHelper`